scripts/gh-detect-changes.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# set -x
+
+CONNECTOR=$1
+
+if [ "$GITHUB_EVENT_NAME" == "pull_request" ]
+then
+  COMPARE_TO="origin/${GITHUB_BASE_REF}"
+else
+  # for non-pr (master) builds, comparison for changes is made against the previous commit,
+  # which might be:
+  # * merge commit, then HEAD^ is the last commit before branching out
+  # * squash commit, then HEAD^ is the previous state of master
+  COMPARE_TO="HEAD^"
+fi
+
+git diff "$COMPARE_TO" --exit-code --quiet "${CONNECTOR}" build.sbt project/
+# git diff "$COMPARE_TO" --exit-code --quiet "${CONNECTOR}" build.sbt project/ .github/workflows/
+DIFF_EXIT_CODE=$?
+
+if [ "$GITHUB_EVENT_NAME" == "schedule" ]
+then
+  echo "Building everything as it is a scheduled build"
+  echo "execute_build=true" >> $GITHUB_ENV
```

Comment:
Will this really work?

From the gh actions docs on the $GITHUB_ENV file:
> This file is unique to the current step and changes for each step in a job

So it isn't actually used in the following steps?

Reply:
Isn't `set-output` the tool to use for passing data between steps? https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter

Reply:
This works as the tiny change in CSV shows (all other connectors builds were short-circuited).

Using outputs is more heavy-weight and allows for passing values across jobs.

Reply:
Here are the docs: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

Reply:
Thanks for clarifying, all good then 👍